### PR TITLE
add methods to get system state from  fullnode

### PR DESCRIPTION
**Pull request by longbowlu**
update: I removed the method to get `SuiSystemState` from authority aggregator because 1. it's simpler to grab that from a fullnode and 2. `SuiSystemState` may change frequently so it may not always work to agree on a state with quorum - it's better to get just the thing you actually want, rather than the entire object. That said, we just just add that function whenever we see a real need.

as title
Note that I replace `AccountAddress` with `SuiAddress` in the relevant structs because of the requirement to implement `JsonSchema` for jsonrpc data. Structurally they are the same.

### CODE REVIEW

File: crates/sui-types/src/object.rs [hunk 1]
```diff
@@ -278,6 +278,14 @@ impl Data {
         }
     }
 
+    pub fn try_into_move(self) -> Option<MoveObject> {
```
Comment (emmazzz) [inline crates/sui-types/src/object.rs:281]: Do we use this function anywhere?

**Comment by emmazzz** [comment]
> That said, we just just add that function whenever we see a real need.

This might be obvious but what is our need for getting the entire system state obj from full node?

**Comment by longbowlu** [comment]
> > That said, we just just add that function whenever we see a real need.
> 
> This might be obvious but what is our need for getting the entire system state obj from full node?

For my current needs, the load gen client wants this for validator committee. I do see other clients will want this as it's the core object on Sui network. An very simple case is an dapp (e.g. explorer) display the current validator information or current staked amounts, next epoch stake amount etc

**Comment by lxfind** [comment]
If load-gen has access to a fullnode, the fullnode should be doing reconfig, and load-gen should not need to be aware of the committee. It seems the case where load-gen needs committee is when you don't actually have a fullnode?

**Comment by longbowlu** [comment]
> If load-gen has access to a fullnode, the fullnode should be doing reconfig, and load-gen should not need to be aware of the committee. It seems the case where load-gen needs committee is when you don't actually have a fullnode?

In the initial version of this PR, I also exposed a grpc endpoint to get `SuiSystemState` in valdiators but I can't think of why load-gen does not have a fullnode. For load-gen's perspective, the simplest route is to connect to a fullnode, and let fullnode reconfig and get `SuiSystemState` from it. Note, load gen would still need the committee info because in one mode it directly submits transactions to validators, not through the full node. In other mode where txes go through fullnode, reconfig should be completely transparent to load-gen.

**Comment by lxfind** [comment]
I am a bit confused, when do you need to use the get_sui_system_state api on fullnode then?

**Comment by longbowlu** [comment]
> I am a bit confused, when do you need to use the get_sui_system_state api on fullnode then?

The plan is to make load-gen to be able to reconfig along with fullnode because for a load-gen we can assume there's a trusted fullnode that it can talk to, at least for today. This would simplify the reconfig logic for load-gen. Then, I'm going to move the current load-gen logic into another place for other clients that do not necessarily have access to a trusted fullnode (or we could simply delete it if we prefer). It would work in this way:
1. load gen submits txes, load gen may notice reconfigs
2. load gen asks a trusted fullnode about the next committee (`SuiSystemState`) until the fullnode finishes reconfig
3. load gen may need to recreate the authority aggregator, if it directly talks to validators to submit txes

You may ask why fullnode exposes an api for `SuiSystemState` but not the committee info? Because we'd like to access to richer information about validators in clients, such as validators name (the human-readable name) and in general `SuiSystemState` is more useful (to dapps) than Committee. However, we could also expose that, if we find that helpful

File: crates/sui-types/src/object.rs (hunk 1 above)
Comment (longbowlu) [reply crates/sui-types/src/object.rs]: good catch, forgot to remove

**Review by patrickkuo — APPROVED** [review]
code LGTM

**Comment by lxfind** [comment]
I am still not following: If there is already a fullnode, why does load-gen needs to reconfig or be aware of the committee? With a fullnode, load-gen should only be submitting tx to the fullnode without having to worry about the committee. I think that we should get the fullnode writepath right first, and make it the default mode for load-gen because that represents the true latency.

**Comment by longbowlu** [comment]
> I am still not following: If there is already a fullnode, why does load-gen needs to reconfig or be aware of the committee? With a fullnode, load-gen should only be submitting tx to the fullnode without having to worry about the committee. I think that we should get the fullnode writepath right first, and make it the default mode for load-gen because that represents the true latency.

In submitting transactions, load-gen can operate in two modes:
1. one is through a fullnode
2. the other is directly talking to validators (in this mode, load-gen can still leverage a trusted fullnode to reconfig by just pulling the SuiSystemState). 
Both are useful in terms of understanding latency breakdown. In fact both are supported now (although #.1 does not support reconfig yet because we need fullnode itself to be able to reconfig, a different story).

I think our discussion digress in this PR since it doesn't really change how load gen reconfig today and what I have in mind will need to be unblocked by fullnode reconfiging. Nevertheless, this PR will be useful in other cases (I'll be sending out another PR soon) so merging it now

**Pull request merged**